docs/fundamentals/environments.rst

```diff
@@ -1,8 +1,120 @@
+
+If you need to check whether the application is running in a particular environment, use ``env.IsEnvironment("environmentname")`` since it will correctly ignore case (instead of checking if ``env.EnvironmentName == "Development"`` for example).
+
+.. literalinclude:: ../../samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:caption: Startup.cs (some parts removed for brevity)
+	:linenos:
+	:dedent: 4
+	:lines: 27-47,89-109,134-135
+	:emphasize-lines: 8,10-15,30-41
+
+The highlighted sections in the example above show several examples of adjusting application configuration based on environment.
+
+In the ``Startup()`` method (constructor), the configuration of the application is set up to optionally allow for environment-specific configuration files (e.g. ``config.Development.json``) that will override any other config settings (because the file is added last in the configuration setup chain - :doc:`learn more about ASP.NET configuration <configuration>`). A call to ``IsEnvironment()`` is also used to ensure that User Secrets are only configured for use in ``Development`` (learn more about User Secrets and the `Secret Manager <https://github.com/aspnet/Home/wiki/DNX-Secret-Configuration>`_).
+
+In ``Configure()``, the environment is checked once more, and if the app is running in a ``Development`` environment, then it enables BrowserLink and error pages (which typically should not be run in production). Otherwise, if the app is not running in a development environment, a standard error handling page is configured to be displayed in response to any unhandled exceptions.
+
+Summary
+-------
+
+ASP.NET 5 provides a number of features and conventions that allow developers to easily control how their applications behave in different environments. When publishing an application from development to staging to production, environment variables set appropriately for the environment allow for optimization of the application for debugging, testing, or production use, as appropriate.
+
+Additional Resources
+--------------------
+
+- `Tag Helpers in ASP.NET MVC 6 <http://docs.asp.net/en/latest/mvc/views/tag-helpers/index.html>`_ including the Environment Tag Helper
+- :doc:`configuration`
+- `Configuring Environment Variables from command / terminal prompt <http://ardalis.com/configuring-asp-net-5-environment-variables>`_ on Windows and Mac OS
```

Comment:
Use of environment variables on various platforms is well document and discoverable on the web, so I think we can remove this link.